rs/nervous_system/tools/sync-with-released-nervous-system-wasms/src/main.rs

```diff
@@ -79,68 +100,100 @@ async fn get_mainnet_canister_git_commit_id_and_module_hash(
             .read_state_canister_info(canister_id, "module_hash")
             .await?
     };
-    let module_hash_str = module_hash.iter().fold(String::new(), |mut output, x| {
-        let _ = write!(output, "{:02x}", x);
-        output
-    });
 
-    Ok((git_commit_id, module_hash_str))
+    Ok((git_commit_id, module_hash_hex(module_hash)))
 }
 
-/// This function fetches the given canister's module hash deployed on the ICP mainnet (based on `canister_id`)
-/// and then finds a git tag in the given `canister_repository` (e.g., `dfinity/cycles-ledger`)
-/// which contains a release asset whose name matches `canister_name`
-/// and whose sha256 hash matches the module hash deployed on the ICP mainnet
-/// (the sha256 hash is fetched from a dedicated release asset whose name has the form `{canister_name}.sha256`).
-/// To find the git tag, the function proceeds as follows:
-///   - it crawls all git tags of the given `canister_repository`;
-///   - for every git tag, it lists all release assets and looks for a release asset whose name has the form `{canister_name}.sha256`:
-///     if that release asset contains the expected module hash (as deployed on the ICP mainnet),
-///     then the corresponding tag is returned.
-async fn get_mainnet_canister_git_tag_and_module_hash(
-    agent: &Agent,
-    canister_id: CanisterId,
-    canister_repository: String,
-    canister_name: String,
-) -> Result<(String, String)> {
-    use std::fmt::Write;
+const GITHUB_API: &str = "https://api.github.com";
```

Comment:
The GitHub API you implement could be useful for others in the mono repo, consider factoring it out into a separate module or even library crate (but let's do this at some later point to save time moving forward with this work).

NCR